app/Http/Controllers/UserController.php

```diff
@@ -50,7 +51,24 @@ public function update(UpdateUserRequest $request)
     {
         $user = $request->user();
         $result = $this->userService->updateUserEmail($request->email, $user);
-        return response()->json(['message' => 'Your email adress successfully updated, new adress is: ' . $result->email], 200);
+        return response()->json(['message' => 'Your email adress successfully updated, new adress is: ' . $result->email],
+            200);
+    }
+
+    public function getUsers()
+    {
+        $users_list = $this->userService->getUsers();
+        // Convert array to list, for easy display
+        $users_list = implode(', ', $users_list);
+        return response()->json(['message' => 'Users: ' . $users_list], 200);
+    }
+
+    public function getUserData(Request $request, $id)
```

Comment:
create custom requests. In authorization method check that user has permission to get data.